src/GovUk.Education.ExploreEducationStatistics.Public.Data.Services.Tests/DataSetVersionPathResolverTests.cs

```diff
@@ -14,12 +14,12 @@
 
 namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Services.Tests;
 
-public abstract class ParquetPathResolverTests
+public abstract class DataSetVersionPathResolverTests
```

Comment:
I see you're making the outer classes abstract and I've seen Nick making them static I think.  Be good to agree upon a best convention.

Reply:
Spoke to @ntsim about this and he has been doing a mixture of both, creating abstract base classes where test classes need to inherit dependencies like the Test application, and in other cases like util tests which don't need to inherit, make the outer class static.

When searching the codebase for classes with the regex `public static class.+Tests` they all seem to be util type tests except for `DataSetFilesControllerCachingTests`, but it's not so clear cut when searching for `public abstract class.+Tests` where there's a mixture of util and non-util type tests.

I've made a change to `DataSetFilesControllerCachingTests`, and two existing processor test classes `CreateInitialDataSetVersionFunctionTests`, and `ProcessInitialDataSetVersionFunctionTests` to make the base classes abstract.

I've also added a note for discussion on the backend test style guide proposals board.

Reply:
Brill thanks for investigating